core/src/consensus/tendermint/mod.rs

```diff
@@ -184,23 +184,27 @@ impl TendermintInner {
 
     /// Get previous block hash to determine validator set
     fn prev_block_hash(&self) -> H256 {
-        self.prev_block_of_height(self.height).hash()
+        self.prev_block_header_of_height(self.height)
+            .expect("Height is increased when previous block is imported")
+            .hash()
     }
 
     /// Get the proposer of previous block to check new proposer is valid.
-    fn prev_block_proposer_idx(&self, height: Height) -> usize {
-        let prev_proposer = self.prev_block_of_height(height).author();
-        self.validators
-            .get_index_by_address(&self.prev_block_hash(), &prev_proposer)
-            .expect("The proposer must be in the validator set")
+    fn prev_block_proposer_idx(&self, height: Height) -> Option<usize> {
+        self.prev_block_header_of_height(height).and_then(|prev_header| {
```

Comment:
Use `map` instead of the `and_then`.
We could remove the unnecessary `Some()`

Reply:
Fixed